### PR TITLE
Add per-column outlier contribution scores (TASKS.md 3.3)

### PR DESCRIPTION
**Pull request by ipeirotis**
$(cat <<'EOF'
## Summary

- **New `compute_per_column_errors()`** in `evaluate/outliers.py` returns shape `(N, num_attrs)` — per-row, per-attribute normalized categorical crossentropy losses, applying the same unseen-category penalty and `log(max(K,2))` normalization as `compute_reconstruction_error`.
- **`compute_reconstruction_error` refactored** to delegate to `compute_per_column_errors().mean(axis=1)` — no behavior change, eliminates duplicated loop logic.
- **`get_outliers_list` extended** with optional `attr_names` parameter; when provided, adds `col_error__{name}` columns to the output DataFrame for each attribute. The aggregate `error` column equals the mean of all `col_error__*` columns.
- **`_compute_attr_layout` extended** to return a third element `attr_names` (original column names in vectorized-matrix slice order). `find_outliers` passes it through to `get_outliers_list` so the CLI output CSV now includes per-column error scores.
- **`compute_per_column_contributions` rewritten** on top of `compute_per_column_errors` — fixes normalization edge case (`log(max(K,2))` vs `max(log(K), 1e-10)`) and adds optional `attr_is_categorical` parameter. All existing `test_contributions.py` tests still pass.
- **18 new tests** in `tests/test_per_column_errors.py` covering output shape/dtype, perfect-reconstruction → zero, aggregate == mean(per_col), worse-attr detection, row independence, DataFrame/ndarray inputs, unseen-category penalty, numeric-attribute non-clamping, `get_outliers_list` col_error__ column presence/absence/values, and `_compute_attr_layout` attr_names correctness.

## Test plan

- [x] All 340 existing tests pass, 12 skipped (no regressions)
- [x] 18 new `test_per_column_errors.py` tests pass
- [x] `test_contributions.py` (6 tests) still passes with refactored `compute_per_column_contributions`
- [x] `test_reconstruction_error.py` (22 tests) still passes

https://claude.ai/code/session_01TBmdErc8baWEWUoeg187jc
EOF
)

### CODE REVIEW

File: evaluate/outliers.py [hunk 1]
```diff
@@ -78,70 +151,10 @@ def compute_reconstruction_error(
...
-            # would conflict with the training loss.
-            ce_final = ce_normalized
-
-        per_attr_losses.append(ce_final)
-        start_idx += categories
-
-    stacked = tf.stack(per_attr_losses, axis=0)  # (num_attrs, N)
-    row_loss = tf.reduce_mean(stacked, axis=0)  # (N,)
-    return np.asarray(row_loss.numpy())
+    return per_col.mean(axis=1)
```
[the remark this answers is not in the document]
Comment (ipeirotis) [reply evaluate/outliers.py]: Fixed in a05887e. Extracted a private `_compute_per_attr_losses_tf` helper that returns the `(num_attrs, N)` TF tensor, and a `_prepare_inputs` helper for validation/casting. `compute_reconstruction_error` now calls `tf.reduce_mean` on the TF tensor before converting — only `(N,)` floats leave TF. `compute_per_column_errors` still transposes the full `(N, num_attrs)` tensor to NumPy when that data is explicitly requested.

---
_Generated by [Claude Code](https://claude.ai/code)_
